src/main/scala/se/nimsa/sbx/box/BoxDAO.scala

```diff
@@ -356,6 +356,13 @@ class BoxDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: ExecutionCont
     boxQuery.drop(startIndex).take(count).result
   }
 
+  def listPendingOutgoingTransactionsForBox(boxId: Long): Future[Seq[OutgoingTransaction]] = db.run {
+    outgoingTransactionQuery
+      .filter(_.boxId === boxId)
+      .filter(_.status === (WAITING: TransactionStatus))
+      .result
+  }
+
```

Comment:
Uses Slick's support for streaming. This means that when sending files to another slicebox, we are streaming all the way from the database (and storage) to the receiver.